benchmark/Benchmark/Compare.hs

```diff
@@ -0,0 +1,45 @@
+module Benchmark.Compare
+       (main
+       ) where
+
+import Data.Text (Text)
+import Gauge.Main (bench, bgroup, defaultMain, whnf)
+import Text.Toml (parseTomlDoc)
+import Text.Toml.Types (Table)
+
+import Benchmark.Htoml (convertHtoml, decodeHtoml)
+import Benchmark.Tomland (codec)
+import Toml.Bi.Code (decode, runTomlCodec)
+import Toml.Parser (parse)
+import Toml.Type (TOML)
+
+import qualified Data.Text.IO as TIO
+
+
+-- | Benchmark.
+main :: IO ()
+main = do
+    (txt, htoml, tomland) <- preload
+    defaultMain
+        [ bgroup "tomland"
+            [ bench "parse"   $ whnf parse txt
+            , bench "convert" $ whnf (runTomlCodec codec) tomland
+            , bench "decode"  $ whnf (decode codec) txt
```

Comment:
I think every module in `Benchmark.*` should export three function: `parse`, `convert` and `decode` so you can just write:

```haskell
bgroup "parse"
    [ bench "tomland" $ nf Tomland.parse txt
    , bench "htoml"   $ nf   Htoml.parse txt
    ...
```